### PR TITLE
Added the Settings Fragment as Preferences.

### LINKED ISSUE #20: Change from BCI to Neuro
**Actual Behaviour**
Change is needed for changing from BCI to Neuro in all instances.
Fixes problems with #19 
Please state here what is currently happening.

**Expected Behaviour**

State here what the feature should enable the user to do.

**Steps to reproduce it**

Add steps to reproduce bugs or add information on the place where the feature should be implemented. Add links to a sample deployment or code.

**LogCat for the issue**

Provide logs for the crash here

**Screenshots of the issue**

Where-ever possible add a screenshot of the issue.

**Would you like to work on the issue?**
Yes I will be working on this issue.

Let us know if this issue should be assigned to you or tell us who you think could help to solve this issue.

### PR DESCRIPTION
**Pull request by jddeep**
Fixes #20 #21 .
This is the beginning for the completion of #17 

**Changes**: Added the Settings Activity with the Settings Fragment along with the required Java file.
Changed the parts where it was required and I was asked for by @CloudyPadmal .

**Screenshot/s for the changes**: 
firstpullreqneurolab

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [ ] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [ ] I have requested reviews from other members

**APK for testing**: [Compress the `app-debug.apk` file into a `<feature>.rar` or `<feature>.zip` file and upload it here]

### CODE REVIEW

**Comment by jddeep** [comment]
@CloudyPadmal @mariobehling  the PR is still showing "Review required". Once this PR is merged I can continue my work in making further Screens(Activities/Fragments) of the Android Application giving the desired functionality with it.

File: README.md [hunk 1]
```diff
@@ -14,3 +14,5 @@ Our brains communicate through neurotransmitters and their activity emits electr
 Current devices in the medical industry are usually not accessible by doctors due to their high pricing. They are also complicated to use. The idea of the device is to integrate it into a headband and focus on signals that can be obtained through the frontal lobe.
 
 A difference to existing projects like OpenBCI is that it will not be necessary to 3D print large headsets. Instead we are focusing on creating a device that collects as much data as possible through the forehead. To achieve this goal we are using high-grade sensors and flexible electronics.
+
+The Android Application need to support from API level 16. Some parts of the desktop version of the application might not support it, so be sure to use alternatives for the same and implement them effectively.
```
Comment (CloudyPadmal) [inline README.md]: No need of this. There is no relation between desktop app features with API16. Two different platforms.

File: app/build.gradle [hunk 2]
```diff
@@ -24,9 +24,10 @@ android {
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation "com.android.support:support-annotations:$rootProject.supportLibraryVersion"
-    implementation "com.android.support:appcompat-v7:$rootProject.supportLibraryVersion"
```
Comment (CloudyPadmal) [inline app/build.gradle]: As stated in the previous review which you said double checked, don't remove appcompact library here.

File: app/build.gradle [hunk 3]
```diff
@@ -24,9 +24,10 @@ android {
...
+    implementation "com.android.support:preference-v7:$rootProject.supportLibraryVersion"
     implementation "com.android.support:design:$rootProject.supportLibraryVersion"
     implementation "com.android.support.constraint:constraint-layout:$rootProject.constraintLayoutVersion"
+    implementation 'com.android.support:preference-v7:26.1.0'
```
Comment (CloudyPadmal) [inline app/build.gradle]: Remove this as it's added in line 27

File: app/src/main/java/io/neurolab/NeuroSettingsFragment.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+        addPreferencesFromResource(R.xml.neuro_settings_fragment);
+        Preference preference = findPreference(getResources().getString(R.string.samples_pref_key));
+        preference.setOnPreferenceChangeListener(this);
+
+        SharedPreferences sharedPreferences = getPreferenceScreen().getSharedPreferences();
+        PreferenceScreen preferenceScreen = getPreferenceScreen();
+        int count = preferenceScreen.getPreferenceCount();
+
+        for(int i=0; i<count; i++){
```
Comment (CloudyPadmal) [inline app/src/main/java/io/neurolab/NeuroSettingsFragment.java]: Reformat the code here to add missing spaces.

File: app/src/main/java/io/neurolab/NeuroSettingsFragment.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+            String value = sharedPreferences.getString(pref.getKey(),"");
+            pref.setSummary(value);
+        }
+    }
+
+    @Override
+    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
+        Preference preference = findPreference(key);
+
```
Comment (CloudyPadmal) [inline app/src/main/java/io/neurolab/NeuroSettingsFragment.java]: Delete extra new line

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]

**Comment by CloudyPadmal** [comment]
Don't close this PR as the previous one. Make changes in the same one and squash commits into one. Thanks.

**Comment by CloudyPadmal** [comment]
You will find https://github.com/fossasia/pslab-android#how-to-git-squash useful

File: README.md (hunk 1 above)
Comment (jddeep) [reply README.md]: No, I know that but for example I was talking about using ConcurrentLinkedDeque whose library could be imported and used in the Desktop version without any issue but in the case of Android the same is not possible for API level 16, we need to annotate it with "Required API level 21". Sir, I hope you get my point.

Comment (jddeep) [reply README.md]: But still @CloudyPadmal sir as you told I have removed this change from this PR.

**Comment by jddeep** [comment]
Thanks @CloudyPadmal sir. All the asked required changes have been seen into and now this PR is mergeable. Happy Contributing!

**Comment by CloudyPadmal** [comment]
Nice work @jddeep ! Could you squash the commits into one?

**Comment by jddeep** [comment]
> Nice work @jddeep ! Could you squash the commits into one?
@CloudyPadmal I Rebased my development branch(from Android Studio) and then commited and pushed one final time. I hope that solves the purpose.

**Comment by jddeep** [comment]
@CloudyPadmal Sir, I followed https://github.com/fossasia/pslab-android#how-to-git-squash and did the reset  and then did ` add . ` , ` commit ` with a extra change with a tag as ` Final commit ` as otherwise, it was showing "No files to commit". Finally, I pushed it to the origin development branch but I cannot actually see the change. It is still showing 6 commits (5+1) which was supposed to show 1 commit after resetting.

**Comment by CloudyPadmal** [comment]
Did you directly use `git remote add upstream https://github.com/fossasia/pslab-android.git` ? It has to be `git remote add upstream https://github.com/fossasia/neurolab-android.git`

**Comment by jddeep** [comment]
> Did you directly use `git remote add upstream https://github.com/fossasia/pslab-android.git` ? It has to be `git remote add upstream https://github.com/fossasia/neurolab-android.git`

No, I used git remote add upstream https://github.com/fossasia/neurolab-android.git only sir.

**Comment by mariobehling** [comment]
Ok, I am squashing it. Please try to learn how to squash. Thank you.

**Pull request merged**